spec/filters/duration_spec.rb

```diff
@@ -0,0 +1,20 @@
+# encoding: utf-8
+require 'spec_helper'
+require "logstash/filters/duration"
+
+describe LogStash::Filters::Duration do
+  describe "Parse duration in ISO format" do
+    let(:config) do <<-CONFIG
+      filter {
+        duration {
+          iso => "P1DT2H3M4S"
+        }
+      }
+    CONFIG
+    end
+
+    sample("iso" => "P1DT2H3M4S") do
```

Comment:
please try to do some more tests.
didn't test
 - negative values
 - just days, without time of day part
 - just time of days without days part
 - milliseconds

the last test could be such an integration test. before try to be as minimal as possible.